packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -72,4 +78,46 @@ export default class Proxy {
     patch(proxy, { address, authenticationToken, name })
     return this._db.update(proxy).then(extractProperties)
   }
+
+  async callProxyMethod(id, method, params, expectStream = false) {
+    const proxy = await this.getProxy(id)
+
+    const { httpRequest } = this._app
+    const response = await httpRequest(proxy.address, {
+      body: format.request(0, method, params),
+      headers: {
+        'Content-Type': 'application/json',
+        Cookie: `authenticationToken=${proxy.authenticationToken}`,
+      },
+      method: 'POST',
+      pathname: '/api/v1',
+    })
+
+    const authenticationToken = setCookie(response, {
+      map: true,
+    })?.authenticationToken?.value
```

Comment:
Why the first `?.`?

Reply:
If the response not contains cookies.

Reply:
Can `set-cookier-parser` return `undefined` or `null` in this case????

Reply:
I'll test which type of result returns in this case generally it should returns a map.